version.go

```diff
@@ -7,58 +7,173 @@ package pool
 
 import (
 	"bytes"
+	"context"
 	"fmt"
+	"math"
 	"strings"
+
+	"google.golang.org/grpc/metadata"
 )
 
 // Commit stores the current commit hash of this build, this should be set
 // using the -ldflags during compilation.
 var Commit string
 
-// semanticAlphabet
+// semanticAlphabet is the allowed characters from the semantic versioning
+// guidelines for pre-release version and build metadata strings. In particular
+// they MUST only contain characters in semanticAlphabet.
 const semanticAlphabet = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-"
 
 // These constants define the application version and follow the semantic
 // versioning 2.0.0 spec (http://semver.org/).
 const (
 	appMajor uint = 0
 	appMinor uint = 4
-	appPatch uint = 2
+	appPatch uint = 3
 
 	// appPreRelease MUST only contain characters from semanticAlphabet per
 	// the semantic versioning spec.
 	appPreRelease = "alpha"
+
+	// defaultAgentName is the default name of the software that is added as
+	// the first part of the user agent string.
+	defaultAgentName = "poold"
+
+	// additionalInfoInitiator is the field name of the initiator in the
+	// additional information part of the user agent.
+	additionalInfoInitiator = "initiator"
 )
 
+// agentName stores the name of the software that is added as the first part of
+// the user agent string. This defaults to the value "poold" when being run as
+// a standalone component but can be overwritten by LiT for example when poold
+// is integrated into the UI.
+var agentName = defaultAgentName
+
+// SetAgentName sets the static part of the user agent string, identifying the
+// type of binary that is running. This function panics if the agent name
+// contains characters outside of the allowed semantic alphabet.
+func SetAgentName(newAgentName string) {
```

Comment:
not used?

Reply:
Yes, it'll be used in LiT to overwrite the default name. Updated the comment to make it clearer.